Build failure: v4l2loopback with testing kernel

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

enable v4l2loopback with testing kernel

Something like

```nix
  boot = {
    kernelPackages = pkgs.linuxPackages_testing;
    extraModulePackages = with config.boot.kernelPackages; [
      v4l2loopback
    ];
    extraModprobeConfig = ''
      options v4l2loopback devices=1 video_nr=1 card_label="OBS Cam" exclusive_caps=1
    '';
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: Cannot build '/nix/store/pfjsmg2ybv7s82vd9j17ca2gcwbgz1cr-v4l2loopback-0.15.1-6.18-rc2.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/pixywza9xp1vrzkm43isv5rvv54ds4w1-v4l2loopback-0.15.1-6.18-rc2-bin
         /nix/store/pv7543wdvh3x439dkskf5rfizwl9ypl9-v4l2loopback-0.15.1-6.18-rc2
       Last 25 log lines:
       >   CC [M]  v4l2loopback.o
       > v4l2loopback.c: In function 'v4l2_loopback_open':
       > v4l2loopback.c:2337:9: error: too few arguments to function 'v4l2_fh_add'
       >  2337 |         v4l2_fh_add(&opener->fh);
       >       |         ^~~~~~~~~~~
       > In file included from /nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/include/media/v4l2-subdev.h:17,
       >                  from /nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/include/media/v4l2-device.h:13,
       >                  from v4l2loopback.c:29:
       > /nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/include/media/v4l2-fh.h:97:6: note: declared here
       >    97 | void v4l2_fh_add(struct v4l2_fh *fh, struct file *filp);
       >       |      ^~~~~~~~~~~
       > v4l2loopback.c: In function 'v4l2_loopback_close':
       > v4l2loopback.c:2386:9: error: too few arguments to function 'v4l2_fh_del'
       >  2386 |         v4l2_fh_del(&opener->fh);
       >       |         ^~~~~~~~~~~
       > /nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/include/media/v4l2-fh.h:125:6: note: declared here
       >   125 | void v4l2_fh_del(struct v4l2_fh *fh, struct file *filp);
       >       |      ^~~~~~~~~~~
       > make[4]: *** [/nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/scripts/Makefile.build:287: v4l2loopback.o] Error 1
       > make[3]: *** [/nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/Makefile:2010: .] Error 2
       > make[2]: *** [/nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/Makefile:248: __sub-make] Error 2
       > make[2]: Leaving directory '/build/source'
       > make[1]: *** [/nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/source/Makefile:248: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/39ygjxsra3ifm57zlsmmz8y7gzyhxran-linux-6.18-rc2-dev/lib/modules/6.18.0-rc2/build'
       > make: *** [Makefile:53: v4l2loopback.ko] Error 2
       For full logs, run:
         nix log /nix/store/pfjsmg2ybv7s82vd9j17ca2gcwbgz1cr-v4l2loopback-0.15.1-6.18-rc2.drv
error: Cannot build '/nix/store/z1rwx7w8mnacmq8yx56rdnc10rak9vf9-linux-6.18-rc2-modules.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/6x35palj09hyb10fvpr565x2516lfh2v-linux-6.18-rc2-modules
error: Cannot build '/nix/store/w75wsmbpfnyykq6q8k32inp1l3q89vpm-nixos-system-Karimihub-25.11.20251020.b3e01bd.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/wnhy06khvq86lsilf4ssw1fikc4q67rv-nixos-system-Karimihub-25.11.20251020.b3e01bd
Traceback (most recent call last):
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/__init__.py", line 358, in main
    execute(sys.argv)
    ~~~~~~~^^^^^^^^^^
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/__init__.py", line 316, in execute
    services.build_and_activate_system(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        action=action,
        ^^^^^^^^^^^^^^
    ...<10 lines>...
        flake_common_flags=flake_common_flags,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/services.py", line 305, in build_and_activate_system
    path_to_config = _build_system(
        attr=attr,
    ...<9 lines>...
        flake_common_flags=flake_common_flags,
    )
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/services.py", line 172, in _build_system
    path_to_config = nix.build_flake(
        attr,
    ...<2 lines>...
        | {"no_link": no_link, "dry_run": dry_run},
    )
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/nix.py", line 87, in build_flake
    r = run_wrapper(run_args, stdout=PIPE)
  File "/nix/store/j2nan8w1nyvjbqxs88qvkqwdicpgmps4-nixos-rebuild-ng-0.0.0/lib/python3.13/site-packages/nixos_rebuild/process.py", line 137, in run_wrapper
    r = subprocess.run(
        run_args,
    ...<7 lines>...
        **kwargs,
    )
  File "/nix/store/62fdlzq1x1ak2lsxp4ij7ip5k9nia3hc-python3-3.13.7/lib/python3.13/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nix', '--extra-experimental-features', 'nix-command flakes', 'build', '--print-out-paths', '/etc/nixos?submodules=1#nixosConfigurations."default".config.system.build.toplevel', '-v', '--no-link']' returned non-zero exit status 1.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.12, NixOS, 25.11 (Xantusia), 25.11.20251018.c38ea86`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/af8n37in1cwnhpppcl2blmlsg46x8lbi-source`

### Notify maintainers

@moni @bot-wxt1221

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/v4l2loopback/default.nix
+++ b/pkgs/os-specific/linux/v4l2loopback/default.nix
@@ -10,13 +10,13 @@
 
 stdenv.mkDerivation rec {
   pname = "v4l2loopback";
-  version = "0.15.1";
+  version = "0.15.3";
 
   src = fetchFromGitHub {
     owner = "umlaeute";
     repo = "v4l2loopback";
     tag = "v${version}";
-    hash = "sha256-uokj0MB6bw4I8q5dVmSO9XMDvh4T7YODBoCCHvEf4v4=";
+    hash = "sha256-KXJgsEJJTr4TG4Ww5HlF42v2F1J+AsHwrllUP1n/7g8=";
   };
 
   hardeningDisable = [

--- a/pkgs/os-specific/linux/v4l2loopback/default.nix
+++ b/pkgs/os-specific/linux/v4l2loopback/default.nix
@@ -5,15 +5,12 @@
   kernel,
   kmod,
   kernelModuleMakeFlags,
+  nix-update-script,
 }:
 
-let
-  version = "0.15.1";
-
-in
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "v4l2loopback";
-  version = "${version}-${kernel.version}";
+  version = "0.15.1";
 
   src = fetchFromGitHub {
     owner = "umlaeute";
@@ -53,6 +50,8 @@ stdenv.mkDerivation {
     "KERNEL_DIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
     "v4l2loopback.ko"
   ];
+
+  passthru.updateScript = nix-update-script { };
 
   meta = {
     description = "Kernel module to create V4L2 loopback devices";

